authentication/permissions.py

```diff
@@ -15,16 +15,32 @@ def has_permission(self, request, view):
         if not request.user.position == self.position:
             raise PermissionDenied(detail=self.error.msg)
         return True
-            
+    
 class IsFounder(PositionPermission):
     position = 'Founder'
     error = Error.NOT_FOUNDER
         
 class IsRepresentative(PositionPermission):
     position = 'Representative'
-    error = Error.NOT_REPRESENTATIVE
+    error = Error.NOT_REPRESENTATIVE  
 
+class CompanyTypePermission(BasePermission):
+    type = None
+    error = None
 
+    def has_permission(self, request, view):
+        if not request.user.get_company_type() == self.type:
```

Comment:
Consider using `!=` for the comparison for better readability: `if request.user.get_company_type() != self.type:`

Reply:
Agree, fixed.